File: usa/update_views.py
import sys

with open("src/views.js", "r", encoding="utf-8") as f:
    content = f.read()

# 1. Remove printBtn
content = content.replace("""    const printBtn = document.getElementById('print-top-trumps-btn-html');
    if (printBtn && !printBtn.dataset.bound) {
      printBtn.addEventListener('click', () => window.print());
      printBtn.dataset.bound = 'true';
    }""", "")

# 2. Update logic for unlocking and iterate with index
# Original: TRADING_CARDS_DATA.forEach(card => {
#           const isUnlocked = forceUnlock || (window.state && window.state.userStats && window.state.userStats.progress && window.state.userStats.progress[card.subtopicId] && window.state.userStats.progress[card.subtopicId].masteryScore >= 100);

new_logic = """    const totalXP = (window.state && window.state.userStats && window.state.userStats.totalXP) || 0;

    TRADING_CARDS_DATA.forEach((card, index) => {
      const requiredXP = (index + 1) * 200;
      const isUnlocked = forceUnlock || totalXP >= requiredXP;"""

old_logic = """    TRADING_CARDS_DATA.forEach(card => {
      const isUnlocked = forceUnlock || (window.state && window.state.userStats && window.state.userStats.progress && window.state.userStats.progress[card.subtopicId] && window.state.userStats.progress[card.subtopicId].masteryScore >= 100);"""

content = content.replace(old_logic, new_logic)

# 3. Update locked text
old_locked_text = """<p style="font-size: 0.85rem;">Master Topic ${card.subtopicId || '...'} to unlock</p>"""
new_locked_text = """<p style="font-size: 0.85rem;">Reach ${requiredXP} XP to unlock</p>"""

content = content.replace(old_locked_text, new_locked_text)

# 4. Make wrapper more faded
old_gradient = "linear-gradient(rgba(10, 10, 15, 0.6), rgba(10, 10, 15, 0.6))"
new_gradient = "linear-gradient(rgba(10, 10, 15, 0.85), rgba(10, 10, 15, 0.85))"

content = content.replace(old_gradient, new_gradient)

with open("src/views.js", "w", encoding="utf-8") as f:
    f.write(content)
print("Updated views.js")

File: cme/update_views.js
const fs = require('fs');
let content = fs.readFileSync('src/views.js', 'utf8');

const targetStr = `export function renderScumbagBinder() {
  const container = document.getElementById("scumbag-binder-grid");
  if (!container) return;
  container.innerHTML = "";

    const totalXP = (window.state && window.state.userStats && window.state.userStats.totalXP) || 0;

  CARDS_DATA.forEach((card, index) => {
    const requiredXP = (index + 1) * 200;
    const isUnlocked = totalXP >= requiredXP;`;

const replaceStr = `export function renderScumbagBinder() {
  const container = document.getElementById("scumbag-binder-grid");
  if (!container) return;
  container.innerHTML = "";

  const totalXP = (window.state && window.state.userStats && window.state.userStats.xp) || 0;
  const unlockedCards = (window.state && window.state.userStats && window.state.userStats.unlockedCards) || [];

  CARDS_DATA.forEach((card, index) => {
    const requiredXP = (index + 1) * 200;
    const hasEnoughXP = totalXP >= requiredXP;
    const isOpened = unlockedCards.includes(card.id);`;

content = content.replace(targetStr, replaceStr);

const targetStr2 = `    if (isUnlocked) {
      frontEl.className += " scumbag-card-unlocked";
      frontEl.style.backgroundImage = \`url('\${card.image}')\`;
      frontEl.innerHTML = '<div class="hologram"></div>';
    } else {
      frontEl.className += " scumbag-card-locked";
      frontEl.innerHTML = \`
        <div class="foil-pack-body" style="position: absolute; inset: 0; background-image: linear-gradient(rgba(10, 10, 15, 0.85), rgba(10, 10, 15, 0.85)), url('assets/mr_lovett_wrapper.png?v=2'); background-size: cover; background-position: center; display: flex; flex-direction: column; align-items: center; justify-content: flex-end; padding: 20px; color: white; text-align: center; border-radius: 12px; overflow: hidden;">
          <i class="fa-solid fa-lock" style="font-size: 2rem; color: #facc15; margin-bottom: 10px;"></i>
          <h3 style="margin-bottom: 5px;">Locked Pack</h3>
          <p style="font-size: 0.85rem;">Reach \${requiredXP} XP to unlock</p>
        </div>
      \`;
    }`;

const replaceStr2 = `    if (hasEnoughXP && isOpened) {
      frontEl.className += " scumbag-card-unlocked";
      frontEl.style.backgroundImage = \`url('\${card.image}')\`;
      frontEl.innerHTML = '<div class="hologram"></div>';
    } else if (hasEnoughXP && !isOpened) {
      frontEl.className += " scumbag-card-ready";
      frontEl.innerHTML = \`
        <div class="foil-pack-body" style="position: absolute; inset: 0; background-image: url('assets/mr_lovett_wrapper.png?v=2'); background-size: cover; background-position: center; display: flex; flex-direction: column; align-items: center; justify-content: flex-end; padding: 20px; color: white; text-align: center; border: 2px solid #facc15; box-shadow: 0 0 15px rgba(250, 204, 21, 0.5); animation: pulse 2s infinite; border-radius: 12px; overflow: hidden; cursor: pointer;">
          <div style="background: rgba(0,0,0,0.8); padding: 15px; border-radius: 8px; width: 100%; border: 1px solid #facc15;">
            <i class="fa-solid fa-gift" style="font-size: 2rem; margin-bottom: 10px; color: #facc15; animation: bounce 1s infinite;"></i>
            <h3 style="margin-bottom: 5px; color: #facc15; text-transform: uppercase;">Tap to Open!</h3>
          </div>
        </div>
      \`;
      frontEl.onclick = () => {
        if (window.triggerPackOpening) window.triggerPackOpening(card.id);
      };
    } else {
      frontEl.className += " scumbag-card-locked";
      frontEl.innerHTML = \`
        <div class="foil-pack-body" style="position: absolute; inset: 0; background-image: linear-gradient(rgba(10, 10, 15, 0.85), rgba(10, 10, 15, 0.85)), url('assets/mr_lovett_wrapper.png?v=2'); background-size: cover; background-position: center; display: flex; flex-direction: column; align-items: center; justify-content: flex-end; padding: 20px; color: white; text-align: center; border-radius: 12px; overflow: hidden;">
          <i class="fa-solid fa-lock" style="font-size: 2rem; color: #facc15; margin-bottom: 10px;"></i>
          <h3 style="margin-bottom: 5px;">Locked Pack</h3>
          <p style="font-size: 0.85rem;">Reach \${requiredXP} XP to unlock</p>
        </div>
      \`;
    }`;

content = content.replace(targetStr2, replaceStr2);

const targetStr3 = `    if (isUnlocked && card.stats) {`;
const replaceStr3 = `    if (hasEnoughXP && isOpened && card.stats) {`;
content = content.replace(targetStr3, replaceStr3);

const targetStr4 = `    if (isUnlocked) {
      wrapperEl.onclick = () => {
        AudioEngine.play('cardFlip');
        innerEl.classList.toggle('flipped');
      };
    }`;
const replaceStr4 = `    if (hasEnoughXP && isOpened) {
      wrapperEl.onclick = () => {
        AudioEngine.play('cardFlip');
        innerEl.classList.toggle('flipped');
      };
    }`;
content = content.replace(targetStr4, replaceStr4);

fs.writeFileSync('src/views.js', content);
console.log('Successfully updated renderScumbagBinder logic in views.js.');
